### PR TITLE
prepare podcast section

### PR DESCRIPTION
**Pull request by houssembaltii**
## Podcast Schema

- **`title`**: The title of the podcast episode.
- **`description`**: A brief description or summary of the episode.
- **`episodeNumber`**: The episode number in the series.
- **`releaseDate`**: The release date of the episode.
- **`duration`**: The duration of the episode in minutes.
- **`podcastUrls`**: An optional array of objects, each with:
  - **`platform`**: The name of the platform (e.g., "Spotify").
  - **`url`**: The URL to the podcast episode on that platform.
- **`coverImageUrl`**: An optional URL of the cover image for the episode.
- **`tags`**: An optional array of tags or categories related to the episode.
- **`hosts`**: an Array of hosts.
- **`guests`**: An array of guests.
- **`language`**: The language in which the episode is recorded (e.g., "french", "english").
- **`notes`**: An optional array of objects, each with([example](https://syntax.fm/show/822/receipt-printer-with-javascript) ):
  - **`time`**: The minute mark of the podcast.
  - **`subject`**: The topic or key point discussed at that minute.

### CODE REVIEW

File: src/content/podcasts/podcasts.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        z.object({
+          platform: z.string(),
+          url: z.string(),
+        }),
+      )
+      .optional(),
+    coverImageUrl: z.string().optional(),
+    tags: z.array(z.string()).optional(),
+    host: z.string(),
+    guests: z.array(z.string()),
```
Comment (JeanneGrenet) [inline src/content/podcasts/podcasts.ts:22]: I think these props can be an object with name and optional picture

**Review by JeanneGrenet — CHANGES_REQUESTED** [review]

File: src/content/podcasts/podcasts.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { z, defineCollection, reference } from "astro:content";
+
+export type Podcast = z.infer<ReturnType<typeof zPodcast>>;
+const zPodcast = () =>
+  z.object({
+    title: z.string(),
+    description: z.string().optional(),
+    episodeNumber: z.number(),
```
Comment (yoannfleurydev) [inline src/content/podcasts/podcasts.ts:8]: Not sure about that one (will be in the episode title I think)

File: src/content/podcasts/podcasts.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    releaseDate: z.date(),
+    duration: z.number(),
+    podcastUrls: z
```
Comment (yoannfleurydev) [inline src/content/podcasts/podcasts.ts:11]:
```suggestion
    urls: z
```

File: src/content/podcasts/podcasts.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      .array(
+        z.object({
+          platform: z.string(),
+          url: z.string(),
```
Comment (yoannfleurydev) [inline src/content/podcasts/podcasts.ts:15]:
```suggestion
          url: z.string().url(),
```

File: src/content/podcasts/podcasts.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  z.object({
+    title: z.string(),
+    description: z.string().optional(),
+    episodeNumber: z.number(),
+    releaseDate: z.date(),
+    duration: z.number(),
+    podcastUrls: z
+      .array(
+        z.object({
+          platform: z.string(),
```
Comment (yoannfleurydev) [inline src/content/podcasts/podcasts.ts:14]:
```suggestion
          name: z.string(),
```

File: src/content/podcasts/podcasts.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    guests: z.array(reference("people")).optional(),
+    language: z.enum(["french", "english"]),
+    notes: z
+      .array(
+        z.object({
+          minute: z.string(),
+          subject: z.string(),
+        }),
+      )
+      .optional(),
```
Comment (yoannfleurydev) [inline src/content/podcasts/podcasts.ts:31]: Can you explain and maybe add a comment to explain that one please?

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

File: src/content/podcasts/podcasts.ts (hunk 6 above)
Comment (houssembaltii) [reply src/content/podcasts/podcasts.ts:31]: I inspired notes field from [syntax fm ](https://syntax.fm/show/822/receipt-printer-with-javascript), where notes will be an array that contains topics and which minute it has been talked about for example : 
* 01:10 : What is Fork It ?
* 2:00 : ForkIt Organizers 
...

**Pull request closed without merging**